tensorflow/contrib/periodic_intersperse/core/kernels/periodic_intersperse_op.h

```diff
@@ -0,0 +1,231 @@
+  {
+    auto last_reduced_i = i;
+    auto r = rank;
+    do {
+      --r;
+      output_indices[r] = last_reduced_i % Y[r];
+      last_reduced_i = (last_reduced_i - output_indices[r]) / Y[r];
+    } while (r > 0);
+  }
+
+  // rasterize the input index
+  {
+    IndexT last_index_factor = 1;
+    for (auto r = rank - 1;; --r) {
+      IndexT index = 0;
+      if (r != q)
+        index = output_indices[r] / g[r];
+      else {
+        for (int qi = 0; qi < rank; ++qi) {
+          if (qi == q) continue;
+          index += G[qi] * (output_indices[qi] % g[qi]);
+        }
+        index *= Y[q];
+        index += output_indices[r];
+      }
+      result += last_index_factor * index;
+      last_index_factor *= X[r];
+      if (r == 0) break;
+    }
+  }
+
+  return result;
+}
+
+template <class T, class VecT>
+void main_logic(tensorflow::OpKernelContext* context, const VecT& desired_shape,
+                const tensorflow::Tensor& input_tensor) {
+  // input is a strided array (last index is fastest, C-ordered)
+  auto input = input_tensor.flat<T>();
+  const int rank = input_tensor.dims();
+  const auto original_size = input.size();
+  // original and target dimensions
+  std::vector<tensorflow::int64> X(rank), Y(rank);
+  tensorflow::int64 total_size = 1, new_sliced_size = 1;
+  // factor by which X increases/decreases w.r.t. Y
+  std::vector<float> f(rank);
+  // helper arrays related to f
+  std::vector<tensorflow::int64> g(rank), G(rank);
+  // index of adjustable dimension
+  int q;
+  tensorflow::TensorShape output_shape;
+
+  // requires that the rank of the input tensor and length of the desired shape
+  // are equal
+  OP_REQUIRES(context, rank == desired_shape.size(),
+              tensorflow::errors::InvalidArgument(
+                  "periodic_intersperse expects the rank of the input tensor, ",
+                  rank, ", to be the same as the length of the desired shape, ",
+                  desired_shape.size(), "."));
+
+  {
+    bool found = false;
+    for (int i = 0; i < rank; ++i) {
+      if (desired_shape(i) < 1) {
+        // only one index can be adjustable
+        OP_REQUIRES(context, !found,
+                    tensorflow::errors::InvalidArgument(
+                        "periodic_intersperse expects only "
+                        "one index to be marked as adjustable."));
+        q = i;
+        found = true;
+      } else {
+        Y[i] = desired_shape(i);
+        new_sliced_size *= Y[i];
+      }
+    }
+    // at least one index needs to be adjustable
+    OP_REQUIRES(context, found, tensorflow::errors::InvalidArgument(
+                                    "periodic_intersperse expects at least "
+                                    "one index to be marked as adjustable."));
+
+    int count = 0;
+    for (const auto dim_info : input_tensor.shape()) {
+      X[count] = dim_info.size;
+      total_size *= X[count];
+      ++count;
+    }
+
+    Y[q] = tensorflow::int64(
```

Comment:
Since you are using floor, isn't this equal to total_size / new_sliced_size (integer division).